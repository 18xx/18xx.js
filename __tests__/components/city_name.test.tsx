import * as React from 'react';
import { ReactElement } from 'react';
import * as renderer from 'react-test-renderer';

import CityName from '../../src/components/city_name';

import Point from '../../src/point';

describe('CityName', () => {
  describe('#render()', () => {
    it('returns an svg representation of the name at a point', () => {
      const subject: ReactElement<CityName> = (
        <CityName
          point={new Point(4, 1)}
          name='Toronto' />
      );

      expect(renderer.create(subject)).toMatchSnapshot();
    });

    describe('when the name is  over 12 characters long', () => {
      it('uses a smaller font size', () => {
        const subject: ReactElement<CityName> = (
          <CityName
            point={new Point(4, 1)}
            name='New Amsterdam' />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });
  });
});
