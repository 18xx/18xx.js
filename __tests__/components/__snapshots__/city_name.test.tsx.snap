// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityName #toString() returns an svg representation of the name at a point 1`] = `
<text
  fontSize={16}
  textAnchor="middle"
  x={4}
  y={1}
>
  Toronto
</text>
`;
